### PR TITLE
[jvm] reduce scala-compiler, scalatest dependency scopes

### PR DESCRIPTION
**Pull request by phraktle**
Related to #6724

### CODE REVIEW

File: jvm-packages/xgboost4j/pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
           <groupId>org.scalatest</groupId>
           <artifactId>scalatest_${scala.binary.version}</artifactId>
           <version>3.0.5</version>
-          <scope>compile</scope>
+          <scope>provided</scope>
```
Comment (CodingCat) [inline jvm-packages/xgboost4j/pom.xml:50]: this should be test ?

Comment (phraktle) [reply jvm-packages/xgboost4j/pom.xml:50]: Ideally, yes. However, the `GpuTestSuite` class references scalatest annotations from the `main` source set.

Comment (CodingCat) [reply jvm-packages/xgboost4j/pom.xml:50]: hmmm? currently the build failed with 

```
[2021-02-24T10:51:04.637Z] [ERROR] /workspace/jvm-packages/xgboost4j-gpu/src/main/java/ml/dmlc/xgboost4j/java/GpuTestSuite.java:[23,21] package org.scalatest does not exist

[2021-02-24T10:51:04.637Z] [ERROR] /workspace/jvm-packages/xgboost4j-gpu/src/main/java/ml/dmlc/xgboost4j/java/GpuTestSuite.java:[25,2] cannot find symbol
```

Comment (phraktle) [reply jvm-packages/xgboost4j/pom.xml:50]: which build? the github CI checks are failing on an unrelated problem in the jenkins stage...

Comment (phraktle) [reply jvm-packages/xgboost4j/pom.xml:50]: Ping...

Comment (phraktle) [reply jvm-packages/xgboost4j/pom.xml:50]: ⏳

**Comment by phraktle** [comment]
All CI jobs are passing.

**Comment by CodingCat** [comment]
thanks for contributing! merged

**Pull request merged**